main.go

```diff
@@ -154,30 +182,56 @@ func (h *Hookbot) IsGithubKeyOK(w http.ResponseWriter, r *http.Request) bool {
 
 	r.Body = ioutil.NopCloser(bytes.NewReader(body))
 
-	mac := hmac.New(sha1.New, []byte(h.github_secret))
-	mac.Reset()
-	mac.Write(body)
+	expected := fmt.Sprintf("sha1=%v", Sha1HMAC(h.github_secret, string(body)))
 
-	signature := fmt.Sprintf("sha1=%x", mac.Sum(nil))
+	return SecureEqual(r.Header.Get("X-Hub-Signature"), expected)
+}
 
-	return SecureEqual(r.Header.Get("X-Hub-Signature"), signature)
+func Sha1HMAC(key, payload string) string {
+	mac := hmac.New(sha1.New, []byte(key))
+	mac.Reset()
```

Comment:
Seems odd to Reset() a fresh hmac. Try removing it and checking we still get the same tokens.